ush/regrid_gsiSfcIncr_to_tile.sh

```diff
@@ -79,63 +64,213 @@ else # construct restart times for deterministic member
+                ${DATA}/vegetation_type.tile${n}.nc"
+    echo "cpreq ${FIXorog}/${CASE_OUT}/${CASE_OUT}_grid.tile${n}.nc \
+                ${DATA}/${CASE_OUT}_grid.tile${n}.nc"
+done
+} > cmdfile.0
 
+for imem in $(seq 1 "${NMEM_REGRID}"); do
+    cmem=$(printf %03i "${imem}")
+    memchar="mem${cmem}"
+
+    # If deterministic job, COMOUT_ATMOS_ANALYSIS_MEM is just COMOUT_ATMOS_ANALYSIS
+    if (( NMEM_REGRID > 1 )); then
         MEMDIR=${memchar} YMD=${PDY} HH=${cyc} declare_from_tmpl \
             COMIN_SOIL_ANALYSIS_MEM:COM_ATMOS_ANALYSIS_TMPL
+
+        memdir="${DATA}/${memchar}"
+        mkdir -p "${memdir}"
+
+        if [[ "${imem}" -gt 1 ]]; then
+            in_dir+=", "
+        fi
+        in_dir+="\"./${memchar}/\""        
+    else
+        # If deterministic job, memdir is just DATA
+        memdir="${DATA}"
+
+        in_dir="'./'"
     fi
 
+    # Create MPMD command file for this member
+    rm -f "cmdfile.${imem}"
+    touch "cmdfile.${imem}"
+    chmod 755 "cmdfile.${imem}"
+
+    # Create commands to stage input files
+    {
+    echo "#!/bin/bash" > "cmdfile.${imem}"
+
     for FHR in "${soilinc_fhrs[@]}"; do
+        echo "cpreq ${COMIN_SOIL_ANALYSIS_MEM}/${APREFIX_ENS}increment.sfc.i00${FHR}.nc \
+                    ${memdir}/sfci00${FHR}.nc"
+    done 
+
+    if [[ "${DO_LAND_IAU}" = ".true." ]]; then 
+        for FHI in "${landifhrs[@]}"; do
+            echo "cpreq ${COMIN_SOIL_ANALYSIS_MEM}/${APREFIX_ENS}increment.sfc.i00${FHI}.nc \
+                        ${memdir}/sfci00${FHI}.nc"
+        done
+    fi
+    } > "cmdfile.${imem}"
+done
 
+# Create master MPMD command file
+rm -f cmdfile
+touch cmdfile
+chmod 755 cmdfile
+
+# Append all members' command files to master command file
+{
+echo "${DATA}/cmdfile.0" # fixed files
+for imem in $(seq 1 "${NMEM_REGRID}"); do
+    echo "${DATA}/cmdfile.${imem}"
+done
+} >> cmdfile
+
+# Run MPMD to stage input files
+"${USHgfs}/run_mpmd.sh" "cmdfile" && true
+export err=$?
+if [[ ${err} -ne 0 ]]; then
+    err_exit "run_mpmd.sh failed!"
+fi
+
+# Finish defining input/output directory list
+export out_dir="${in_dir}"
+
+#
+# Regrid soil increments and save to COMOUT
+#
+
+# Increments for offline analysis
+# If land IAU --> deterministic only. If no land IAU --> both deterministic and ensemble
+if [[ "${DO_LAND_IAU}" = ".false." || "${RUN}" == "gdas" || "${RUN}" == "gfs" ]]; then
+    for FHR in "${soilinc_fhrs[@]}"; do
+        # Set namelist variables
         export add_time_dim=".false."
         export time_list="${FHR}"
+        export out_fname="'sfci00${FHR}'"
 
+        # Create regrid namelist
         rm -f "regrid.nml"
         atparse < "${regrid_nml_tmpl}" >> "regrid.nml"
 
-        cpreq "${COMIN_SOIL_ANALYSIS_MEM}/${APREFIX_ENS}increment.sfc.i00${FHR}.nc" \
-               "${DATA}/enkfgdas.sfci00${FHR}.nc"
+        # Run regrid executable
+        ${APRUN_REGRID} "${REGRID_EXEC}" "1>${PGMOUT}" "2>${PGMERR}"
```

Comment:
The `1>` and `2>` should not be in quotes.  This will be treated as an argument rather than a redirect.

```suggestion
        ${APRUN_REGRID} "${REGRID_EXEC}" 1>"${PGMOUT}" 2>"${PGMERR}"
```